tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/decisiontree/RandomForest.scala

```diff
@@ -80,36 +86,38 @@ class RandomForestTrainer(validationPercentage: Double, numDecisionTrees: Int,
   maximumDepthPerTree: Int = Integer.MAX_VALUE)
     extends ProbabilisticClassifierTrainer {
 
+  private val dtTrainer = new DecisionTreeTrainer(validationPercentage, featuresExaminedPerNode,
+    maximumDepth = maximumDepthPerTree)
+
   /** Induces a RandomForest from a set of feature vectors.
     *
     * @param data a set of feature vectors to use for training
     * @return the induced random forest
     */
   override def apply(data: FeatureVectorSource): ProbabilisticClassifier = {
```

Comment:
Looking at it right now. Is it possible that `DecisionTree.mostProbablyOutcome` and `DecisionTree.distribution` are big, and by serializing and deserializing, you strip that info?

If that's the case, we can probably find an easier way to do this. Like, making them `def`s instead of `@transient lazy val`s.